numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -1927,17 +1927,26 @@ PyArray_Zero(PyArrayObject *arr)
         return NULL;
     }
 
-    obj=PyInt_FromLong((long) 0);
+    if (zero_obj == NULL) {
+        zero_obj = PyInt_FromLong((long) 0);
+        if (zero_obj == NULL) {
+            return NULL;
+        }
+    }
     if (PyArray_ISOBJECT(arr)) {
```

Comment:
> `[zero]val` is a valid pointer to a block of memory large enough to hold `sizeof(PyIntObject)`, but it is actually a `void* NULL`

citation needed. I see a `zeroval = PyDataMem_NEW(PyArray_DESCR(arr)->elsize);` on the line above